ui/src/components/Forms/BurnForm.tsx

```diff
@@ -93,11 +93,16 @@ export const BurnForm: React.FC = () => {
   const handleTokenIndexChange = (
     event: React.ChangeEvent<HTMLInputElement>
   ) => {
-    setTokenIndex(event.target.value);
+    const index = parseInt(event.target.value);
```

Comment:
We can't technically rely on tokenIndex being an integer. Our current token connectors behave that way, but it is a string field in our tokens spec.

Reply:
I think the main problem would be like... it's a number but it's a massive one that JS can't hold as int. But _technically_ it can be any string.

Reply:
oooh i didn't know that, will switch back